### PR TITLE
Add upgrade info

### PR DESCRIPTION
**Pull request by dbrattli**
- Add upgrade info from v2 to v3
- Loosen `choose` requirement from list to seq (which works just as well when using lists).
- Format code
- Add Fantomas to GitHub actions.
- Add GitHub example
- Use Ply instead of Taskbuilder (it's faster)
- User FsToolkit.Errorhandling instead of own Result overrides

### CODE REVIEW

File: paket.dependencies [hunk 1]
```diff
@@ -46,15 +47,15 @@ group Google
...
+    nuget Ply
     nuget Oryx
 
 group Test
     source https://www.nuget.org/api/v2
     storage: none
 
     nuget FSharp.Core >= 4.7
-    nuget coverlet.msbuild 2.5.1
+    nuget coverlet.msbuild ~> 3
```
Comment (dbrattli) [inline paket.dependencies:58]: Note that this is the reason for the code coverage being reduced. We will need to disable coverage requrement on GitHub to get it merged

File: README.md [hunk 2]
```diff
@@ -604,6 +604,52 @@ type Context<'T> =
     }
 ```
 
+## Upgrade from Oryx v2 to v3
+
+Oryx v3 is mostly backwards compatible with v2. Your chains of operators will for most part look and work exactly the
+same. There are however some notable changes:
+
+- The `retry` operator have been deprecated for now. Use [Polly](https://github.com/App-vNext/Polly) instead.
```
Comment (polomani) [inline README.md:612]:
```suggestion
- The `retry` operator has been deprecated for now. Use [Polly](https://github.com/App-vNext/Polly) instead.
```

File: examples/github/GitHub.fsproj [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project Sdk="Microsoft.NET.Sdk.Web">
+    <PropertyGroup>
+        <OutputType>Exe</OutputType>
+        <TargetFramework>net5</TargetFramework>
+    </PropertyGroup>
+    <ItemGroup>
+        <PackageReference Include="Oryx" Version="3.0.0-alpha-002" />
+        <PackageReference Include="Oryx.ThothJsonNet" Version="3.0.0-alpha-002" />
```
Comment (polomani) [inline examples/github/GitHub.fsproj:9]:
```suggestion
        <PackageReference Include="Oryx.ThothJsonNet" Version="3.0.0-beta-001" />
```

File: examples/github/GitHub.fsproj [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project Sdk="Microsoft.NET.Sdk.Web">
+    <PropertyGroup>
+        <OutputType>Exe</OutputType>
+        <TargetFramework>net5</TargetFramework>
+    </PropertyGroup>
+    <ItemGroup>
+        <PackageReference Include="Oryx" Version="3.0.0-alpha-002" />
```
Comment (polomani) [inline examples/github/GitHub.fsproj:8]:
```suggestion
        <PackageReference Include="Oryx" Version="3.0.0-beta-001" />
```

File: test/Fetch.fs [hunk 5]
```diff
@@ -105,93 +106,6 @@ let ``Post url encoded with return expression is Ok`` () =
         test <@ urldecoded'.Contains "foo=bar" @>
     }
 
-// [<Fact>]
```
Comment (polomani) [inline test/Fetch.fs:108]: why this was commented out?

Comment (dbrattli) [reply test/Fetch.fs:108]: Retry not supported anymore so we're using Polly instead.

**Review by polomani — APPROVED** [review]

**Pull request merged**